### PR TITLE
Check Tests

### PR DESCRIPTION
**Pull request by AFH7233**

### CODE REVIEW

File: product-crud/src/test/java/com/afh/skytouch/services/product/queue/producers/FindAllMessageSenderTest.java [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+import com.afh.skytouch.commons.dto.FindAllMessage;
+import com.afh.skytouch.commons.dto.GenericProduct;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.springframework.amqp.rabbit.core.RabbitTemplate;
+
+import static org.junit.Assert.*;
```
Comment (grunger94) [inline product-crud/src/test/java/com/afh/skytouch/services/product/queue/producers/FindAllMessageSenderTest.java]: Avoid using wildcard imports

File: commons-library/pom.xml [hunk 2]
```diff
@@ -30,24 +29,38 @@
         <dependency>
             <groupId>org.projectlombok</groupId>
             <artifactId>lombok</artifactId>
-            <version>${lombok.version}</version>
```
Comment (grunger94) [inline commons-library/pom.xml]: Why were these versions brought back?

File: docker/docker-compose.yml [hunk 3]
```diff
@@ -0,0 +1,25 @@
+version: '3'
+
+services:
+  rabbitmq:
+    container_name: rabbitmq
+    hostname: rabbit
+    image: rabbitmq:management
+    ports:
+      - "5672:5672"
```
Comment (grunger94) [inline docker/docker-compose.yml]: The default port should be overridden

File: commons-library/pom.xml [hunk 4]
```diff
@@ -30,24 +29,38 @@
...
             <groupId>org.hibernate.javax.persistence</groupId>
             <artifactId>hibernate-jpa-2.1-api</artifactId>
-            <version>${hibernate.version}</version>
+            <version>1.0.2.Final</version>
         </dependency>
-
         <dependency>
-            <groupId>org.springframework.boot</groupId>
-            <artifactId>spring-boot-starter-amqp</artifactId>
+            <groupId>org.jetbrains</groupId>
```
Comment (grunger94) [inline commons-library/pom.xml]: Did these dependencies sneak in by mistake?

File: management-api/src/main/java/com/afh/skytouch/managment/controllers/GenericProductController.java [hunk 5]
```diff
@@ -1,52 +1,149 @@
...
 
 @Controller
 @RequestMapping("/genericProduct")
 public class GenericProductController {
 
-    private static final String PRODUCT_ATTRIBUTE = "product";
-    private static final String CREATE = "create-product";
-    private static final String HOME = "home";
-    private static final String SHOW_PRODUCTS = "list-products";
+    public static final String PRODUCT_ATRIBUTE_NAME = "product";
```
Comment (grunger94) [inline management-api/src/main/java/com/afh/skytouch/managment/controllers/GenericProductController.java]: Could these constants be private?

File: management-api/src/main/java/com/afh/skytouch/managment/controllers/GenericProductController.java [hunk 6]
```diff
@@ -1,52 +1,149 @@
...
+    public void setCreationInbox(QueueInbox<String,ProductStatus> creationInbox){
+        this.creationInbox = creationInbox;
+    }
+
+    @Autowired
+    public void setProductListInbox(QueueInbox<String, FindAllMessage>  productListInbox){
+        this.productListInbox = productListInbox;
+    }
+
+    @Value("${managment.service.timeout}")
```
Comment (grunger94) [inline management-api/src/main/java/com/afh/skytouch/managment/controllers/GenericProductController.java]: Typo in management?

File: management-api/src/main/java/com/afh/skytouch/managment/controllers/GenericProductController.java [hunk 7]
```diff
@@ -1,52 +1,149 @@
...
+        this.allMessageSender = allMessageSender;
     }
 
     @GetMapping("/showProduct")
-    public String getProducts(Model model){
-        return SHOW_PRODUCTS;
+    public String getProducts(@ModelAttribute("id") String id,
+                              @ModelAttribute("time") String time,
+                              Model model){
+        System.out.println("Checking again: " + id);
```
Comment (grunger94) [inline management-api/src/main/java/com/afh/skytouch/managment/controllers/GenericProductController.java]: Remove any prints left behind

File: commons-library/src/test/java/com/afh/skytouch/commons/dto/FindAllMessageTest.java [hunk 8]
```diff
@@ -0,0 +1,50 @@
...
+        message.addAll(products);
+        assertEquals(products,message.getProducts());
+    }
+
+    @Test
+    public void testEquals() throws NoSuchFieldException, IllegalAccessException {
+        FindAllMessage messageA = new FindAllMessage();
+        FindAllMessage messageB = new FindAllMessage();
+        Field id = messageB.getClass().getDeclaredField("id");
+        id.setAccessible(true);
```
Comment (grunger94) [inline commons-library/src/test/java/com/afh/skytouch/commons/dto/FindAllMessageTest.java]: Avoid using reflection for unit tests. Investigate a bit about PowerMock's Whitebox.

**Review by grunger94 — CHANGES_REQUESTED** [review]

File: product-crud/src/test/java/com/afh/skytouch/services/product/queue/producers/FindAllMessageSenderTest.java (hunk 1 above)
Comment (AFH7233) [reply product-crud/src/test/java/com/afh/skytouch/services/product/queue/producers/FindAllMessageSenderTest.java]: Fixed

File: commons-library/pom.xml (hunk 2 above)
Comment (AFH7233) [reply commons-library/pom.xml]: It had a bug related to the naming and then i forgot to brought it back. Fixed now

File: docker/docker-compose.yml (hunk 3 above)
Comment (AFH7233) [reply docker/docker-compose.yml]: Changed

File: commons-library/pom.xml (hunk 4 above)
Comment (AFH7233) [reply commons-library/pom.xml]: removed, but I think intelliJ added them.

File: management-api/src/main/java/com/afh/skytouch/managment/controllers/GenericProductController.java (hunk 5 above)
Comment (AFH7233) [reply management-api/src/main/java/com/afh/skytouch/managment/controllers/GenericProductController.java]: They were but I reused them in the test, so I think at least could be package private.

File: management-api/src/main/java/com/afh/skytouch/managment/controllers/GenericProductController.java (hunk 6 above)
Comment (AFH7233) [reply management-api/src/main/java/com/afh/skytouch/managment/controllers/GenericProductController.java]: jejeje yes it was a typo thanks

File: management-api/src/main/java/com/afh/skytouch/managment/controllers/GenericProductController.java (hunk 7 above)
Comment (AFH7233) [reply management-api/src/main/java/com/afh/skytouch/managment/controllers/GenericProductController.java]: Removed

File: commons-library/src/test/java/com/afh/skytouch/commons/dto/FindAllMessageTest.java (hunk 8 above)
Comment (AFH7233) [reply commons-library/src/test/java/com/afh/skytouch/commons/dto/FindAllMessageTest.java]: Changed to use whitebox

**Pull request merged**